### PR TITLE
The Vapouring/Smigglerettening REBORN

### PR DESCRIPTION
**Pull request by TheRealScarHomie**
## Vapour

Ports pollution and renames it to vapour from uhhh, here:
https://github.com/tgstation/tgstation/pull/61154

Currently affects fires, barrel fires, corpses and various smells, the baseline for future expansion. Also adds in a new turf fire system thatll be moved too in the future.

No more indoor firepits, you will DIE if the area is not properly ventilated. It even travels up Z levels n shit.

Bodies give off miasma after about an hour, corpse disposal is key.

Numbers are fully tweaked and GOOD TO GO

image
image

## Cigarettes

A massive overhaul and implementation to cigarettes, with complete new functionality, states and usage:
A Brief rundown of what they do,
You gain benefits for smoking per cigarette, after its effects run out you suffer the comedown effects. 
Chain smoking is not a good idea (just like real life!!)
Different types of smigglerettes have slightly different effects

- [x] cigarettes
- [x] packs
- [x] nicotine effects/cigarette effects
- [x] craftable rollies, dryable produce to use
- [x] spawners, cigarette vendor functionality
- [x] lighters, matches, lighter fluid, etc.

image

guwienko

## Other

scissors??? rope??? drying racks??? tons of random stuff and fixes, absolute carnage to atomisation

### CODE REVIEW

**Comment by Hekzder** [comment]
cereal guy 
image

**Comment by ChungusGamer666** [comment]
cap🧢

**Comment by ProfessorPopoff** [comment]
inhale sounds are comically loud, same with the ciggie burning in general. cut that down 70% nshh...

**Comment by TheRealScarHomie** [comment]
## Okay its actually done now guys

**Comment by ProfessorPopoff** [comment]
Dropping stuff here so you don't forget :)

Drying is iffy for some things. Drying doesn't seem to ever finish in general, and you can't hang Xander up to dry, even though I see you had it as a subtype.

There's a runtime when you fish out cigarettes from a vending machine

Cigarettes should have nicotine overdose toned back a LITTLE BIT maybe. But we'll see in full tests.

Washing paper doesn't actually give you rolling paper

more tbd

**Comment by ProfessorPopoff** [comment]
There's currently a really big issue where the visuals of a vapour, or at least the one visible one we have right now, will fluctuate its visuals rapidly. I thought for a while that it was despawning itself and spawning back in, but you can go into a room that is half empty of smoke and start choking to death..
3532b16208ce66dbb83d667129e80c0b

**Comment by Hekzder** [comment]
whoa

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -108,11 +115,21 @@
...
 
 			else if(isturf(loc)) //Breathe from loc as turf
+				//MOJAVE SUN EDIT START - Vapour
+				var/turf/our_turf = loc
+				if(isopenturf(our_turf))
+					var/turf/open/open_turf = our_turf
+					if(open_turf.vapour)
+						if(next_smell <= world.time)
+							next_smell = world.time + 1 MINUTES //SMELL_COOLDOWN
+							open_turf.vapour.SmellAct(src)
```
Comment (ChungusGamer666) [inline code/modules/mob/living/carbon/life.dm:125]: imagine the smell

File: mojave/__DEFINES/vapour.dm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#define VAPOUR_HEIGHT_DIVISOR 10
+
+#define TICKS_TO_DISSIPATE 6
+
+#define VAPOUR_TASK_PROCESS 1
+#define VAPOUR_TASK_DISSIPATE 2
+
+#define SCENT_DESC_ODOR        "odour"
+#define SCENT_DESC_SMELL       "smell"
+#define SCENT_DESC_FRAGRANCE   "fragrance"
```
Comment (ChungusGamer666) [inline mojave/__DEFINES/vapour.dm:17]: imagine the fragrance

File: code/datums/looping_sounds/_looping_sound.dm [hunk 3]
```diff
@@ -39,7 +39,7 @@
 
 /datum/looping_sound/New(_parent, start_immediately=FALSE, _direct=FALSE, _skip_starting_sounds = FALSE, _channel = 0)  // MOJAVE MODULE OUTDOOR_EFFECTS
 	if(!mid_sounds)
-		WARNING("A looping sound datum was created without sounds to play.")
+		//WARNING("A looping sound datum was created without sounds to play.") MOJAVE SUN EDIT - feature
```
Comment (ChungusGamer666) [inline code/datums/looping_sounds/_looping_sound.dm:42]: it's a feature!

**Comment by Hekzder** [comment]
This is so freaking awesome

**Pull request merged**